### PR TITLE
Fix OSError on _nc_dataset method

### LINKED ISSUE #261: `to_xarray` method error on OS X
Hi,

As described in #260, I am getting an error when trying to use the `to_xarray` method on OS X. Here is the code to reproduce the error:
```python
In [1]: from erddapy import ERDDAP

In [2]: e = ERDDAP(
   ...:         server="https://standards.sensors.ioos.us/erddap/",
   ...:         response="htmlTable",
   ...:     )

In [3]: e.dataset_id = "org_cormp_cap2"

In [4]: e.protocol = "tabledap"

In [5]: e.variables = ["sea_water_temperature", "time"]

In [6]: e.constraints = {
   ...:         "time>=": "1999-10-01T00:00:00Z",
   ...:         "time<=": "1999-10-01T21:00:00Z",
   ...:         "latitude>=": 30,
   ...:         "latitude<=": 33,
   ...:         "longitude>=": -80,
   ...:         "longitude<=": -78,
   ...:     }
```

`to_pandas()` and `to_iris()` work (although I get some UserWarnings on the latter, not sure if they are expected):

```python
In [7]: e.to_pandas()
Out[7]: 
    sea_water_temperature (degree_Celsius)            time (UTC)
0                                    27.30  1999-10-01T00:08:00Z
1                                    27.30  1999-10-01T01:08:00Z
2                                    27.26  1999-10-01T02:08:00Z
3                                    27.24  1999-10-01T03:08:00Z
4                                    27.27  1999-10-01T04:08:00Z
5                                    27.32  1999-10-01T05:08:00Z
6                                    27.34  1999-10-01T06:08:00Z
7                                    27.35  1999-10-01T07:08:00Z
8                                    27.36  1999-10-01T08:08:00Z
9                                    27.33  1999-10-01T09:08:00Z
10                                   27.30  1999-10-01T10:08:00Z
11                                   27.31  1999-10-01T11:08:00Z
12                                   27.30  1999-10-01T12:08:00Z
13                                   27.28  1999-10-01T13:08:00Z
14                                   27.29  1999-10-01T14:08:00Z
15                                   27.31  1999-10-01T15:08:00Z
16                                   27.37  1999-10-01T16:08:00Z
17                                   27.42  1999-10-01T17:08:00Z
18                                   27.49  1999-10-01T18:08:00Z
19                                   27.56  1999-10-01T19:08:00Z
20                                   27.60  1999-10-01T20:08:00Z

In [8]: e.to_iris()
/Users/vwsalazar/miniconda3/envs/erddapy/lib/python3.10/site-packages/iris/fileformats/cf.py:283: UserWarning: Missing CF-netCDF ancillary data variable 'sea_water_temperature_qc_agg', referenced by netCDF variable 'sea_water_temperature'
  warnings.warn(message % (name, nc_var_name))
/Users/vwsalazar/miniconda3/envs/erddapy/lib/python3.10/site-packages/iris/fileformats/cf.py:283: UserWarning: Missing CF-netCDF ancillary data variable 'sea_water_temperature_qc_tests', referenced by netCDF variable 'sea_water_temperature'
  warnings.warn(message % (name, nc_var_name))
/Users/vwsalazar/miniconda3/envs/erddapy/lib/python3.10/site-packages/iris/fileformats/cf.py:1189: UserWarning: Ignoring variable 'latitude' referenced by variable 'sea_water_temperature': Dimensions ('timeseries',) do not span ('obs',)
  warnings.warn(msg)
/Users/vwsalazar/miniconda3/envs/erddapy/lib/python3.10/site-packages/iris/fileformats/cf.py:1189: UserWarning: Ignoring variable 'longitude' referenced by variable 'sea_water_temperature': Dimensions ('timeseries',) do not span ('obs',)
  warnings.warn(msg)
Out[8]: 
[<iris 'Cube' of (41029 / CAP2) Capers Nearshore / (unknown) (-- : 1)>,
<iris 'Cube' of Number of Observations for this TimeSeries / (unknown) (-- : 1)>,
<iris 'Cube' of sea_water_temperature / (degree_Celsius) (-- : 21)>,
<iris 'Cube' of longitude / (degrees) (-- : 1)>,
<iris 'Cube' of latitude / (degrees) (-- : 1)>]
```

The `to_xarray` method, however, returns the following error:

```python
In [9]: e.to_xarray()
---------------------------------------------------------------------------
PermissionError                           Traceback (most recent call last)
File ~/Bio/phd/erddapy/erddapy/netcdf_handling.py:18, in _nc_dataset(url, auth, **requests_kwargs)
     17 try:
---> 18     return Dataset(Path(urlparse(url).path).name, memory=data.read())
     19 except OSError:
     20     # if libnetcdf is not compiled with in-memory support fallback to a local tmp file

File src/netCDF4/_netCDF4.pyx:2353, in netCDF4._netCDF4.Dataset.__init__()

File src/netCDF4/_netCDF4.pyx:1963, in netCDF4._netCDF4._ensure_nc_success()

PermissionError: [Errno 1] Operation not permitted: b'org_cormp_cap2.ncCF'

During handling of the above exception, another exception occurred:

OSError                                   Traceback (most recent call last)
Input In [9], in <cell line: 1>()
----> 1 e.to_xarray()

File ~/Bio/phd/erddapy/erddapy/erddapy.py:615, in ERDDAP.to_xarray(self, **kw)
    613 response = "nc" if self.protocol == "griddap" else "ncCF"
    614 url = self.get_download_url(response=response)
--> 615 nc = _nc_dataset(url, auth=self.auth, **self.requests_kwargs)
    616 return xr.open_dataset(xr.backends.NetCDF4DataStore(nc), **kw)

File ~/Bio/phd/erddapy/erddapy/netcdf_handling.py:22, in _nc_dataset(url, auth, **requests_kwargs)
     19 except OSError:
     20     # if libnetcdf is not compiled with in-memory support fallback to a local tmp file
     21     with _tempnc(data) as _nc:
---> 22         return Dataset(_nc)

File src/netCDF4/_netCDF4.pyx:2353, in netCDF4._netCDF4.Dataset.__init__()

File src/netCDF4/_netCDF4.pyx:1963, in netCDF4._netCDF4._ensure_nc_success()

OSError: [Errno -51] NetCDF: Unknown file format: b'/var/folders/km/_h3vxd093rj42pw0_pk16xdr0000gn/T/erddapy_xr2eatdj.nc'
```

Looks like it drops into the `except OSError` of `_nc_dataset`, and there is a `PermissionError` when trying to write the tempfile. I'm thinking it may be related to [this issue](https://github.com/Unidata/netcdf4-python/issues/1125).

The error doesn't happen with a different server/dataset:

```python
In [10]: e.server = "https://erddap.sensors.ioos.us/erddap/"

In [11]: e.dataset_id = "osmc_23091"

In [12]: e.constraints = {
    ...:         "time>=": "2018-10-01T00:00:00Z",
    ...:         "time<=": "2018-10-01T21:00:00Z",
    ...:         "latitude>=": 10,
    ...:         "latitude<=": 20,
    ...:         "longitude>=": 80,
    ...:         "longitude<=": 90,
    ...:     }

In [13]: e.to_xarray()
Out[13]: 
<xarray.Dataset>
Dimensions:                (timeseries: 1, obs: 8)
Coordinates:
    longitude              (timeseries) float64 ...
    latitude               (timeseries) float64 ...
    time                   (obs) datetime64[ns] ...
Dimensions without coordinates: timeseries, obs
Data variables:
    station                (timeseries) object ...
    rowSize                (timeseries) int32 ...
    sea_water_temperature  (obs) float64 ...
Attributes: (12/53)
    cdm_data_type:                 TimeSeries
    cdm_timeseries_variables:      station,longitude,latitude
    contributor_email:             feedback@axiomdatascience.com
    contributor_name:              Axiom Data Science
    contributor_role:              processor
    contributor_role_vocabulary:   NERC
    ...                            ...
    standard_name_vocabulary:      CF Standard Name Table v72
    summary:                       Timeseries data from '23091 - Moored Buoy'...
    time_coverage_end:             2018-10-01T21:00:00Z
    time_coverage_start:           2018-10-01T00:00:00Z
    title:                         23091 - Moored Buoy
    Westernmost_Easting:           89.203
```

Any idea of what might be happening? Not sure if the error is caused by a problem in the `org_cormp_cap2` dataset, if the PermissionError is a OS X-specific problem, or if there's something else I am missing.

Thank you,
Vini

### PR DESCRIPTION
**Pull request by vinisalazar**
Hi,

this is in relation to #261.

**Summary of changes**
  - Call data.seek(0) after reading 'data' object for the first time (Closes #261)
  - Update 'test_to_objects' to use gold standard Sensors server from #258
  - Rewrite cassette

Thanks,
V

### CODE REVIEW

File: tests/test_to_objects.py [hunk 1]
```diff
@@ -2,29 +2,36 @@
...
 
+import dask
 import httpx
 import iris
 import pytest
 import xarray as xr
 
 from erddapy import ERDDAP
 
+dask.config.set(scheduler="single-threaded")
```
Comment (ocefpaf) [inline tests/test_to_objects.py:13]: This is a better workaround for the netcdf 1.6.1 problem. Netcdf-c is not thread safe and we should not rely on it.

File: erddapy/core/netcdf.py [hunk 2]
```diff
@@ -29,10 +30,17 @@ def _tempnc(data: BinaryIO) -> Generator[str, None, None]:
 
     tmp = None
     try:
-        tmp = NamedTemporaryFile(suffix=".nc", prefix="erddapy_")
+        from sys import platform
+
+        # Delete flag must be False when using Windows
+        # The file will be closed in the 'finally' block
+        delete = False if platform == "win32" else True
```
Comment (ocefpaf) [inline erddapy/core/netcdf.py:37]: Looks like this is not working. Was this failing on Windows before?

Comment (vinisalazar) [reply erddapy/core/netcdf.py:37]: The CI started failing when I introduced the `data.seek(0)` line in [061bf1a](https://github.com/ioos/erddapy/pull/271/commits/061bf1ac8f870bd2fea3c8f0db2c88429e7927c6)

**Comment by ocefpaf** [comment]
@vinisalazar in light of #272 do we still need this one?

**Comment by vinisalazar** [comment]
I don't think we do. Thanks!

**Pull request closed without merging**